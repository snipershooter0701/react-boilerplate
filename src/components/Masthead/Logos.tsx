import React from 'react';
import styled from 'styled-components/macro';
import { ReactComponent as CRALogo } from './cra-logo.svg';
import { ReactComponent as RPLogo } from './rp-logo.svg';
import { ReactComponent as PlusSign } from './plus-sign.svg';

export default function Logos() {
  return (
    <Wrapper>
      <CRALogo className="logo" />
      <PlusSign className="sign" />
      <RPLogo className="logo" />
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;

  .logo {
    width: 4.5rem;
    height: 4.5rem;
  }

  .sign {
    width: 2rem;
    height: 2rem;
    margin: 0 2rem;
  }
`;
