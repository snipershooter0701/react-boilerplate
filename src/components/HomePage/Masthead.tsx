import React from 'react';
import styled from 'styled-components/macro';
import Logos from './Logos';
import Title from 'components/Title';
import Lead from 'components/Lead';
import A from 'components/A';

export default function Masthead() {
  return (
    <Wrapper>
      <Logos />
      <Title>React Boilerplate meets CRA</Title>
      <Lead>
        Now you can use the{' '}
        <A
          href="https://www.reactboilerplate.com/"
          target="_blank"
          rel="noopener noreferrer"
        >
          React Boilerplate
        </A>{' '}
        as{' '}
        <A
          href="https://create-react-app.dev/"
          target="_blank"
          rel="noopener noreferrer"
        >
          Create React App
        </A>{' '}
        template.
      </Lead>
    </Wrapper>
  );
}

const Wrapper = styled.main`
  height: 60vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 320px;
`;
