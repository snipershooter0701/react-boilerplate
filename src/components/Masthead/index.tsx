import React from 'react';
import styled from 'styled-components/macro';
import Container from 'components/Container';
import Logos from './Logos';
import Title from 'components/Title';
import Lead from 'components/Lead';
import A from 'components/A';

export default function Masthead() {
  return (
    <Wrapper>
      <Container>
        <Logos />
        <Title>React Boilerplate meets CRA</Title>
        <Lead>
          Now you can use the{' '}
          <A
            href="https://www.reactboilerplate.com/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React Boilerplate
          </A>{' '}
          as{' '}
          <A
            href="https://create-react-app.dev/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Create React App
          </A>{' '}
          template.
        </Lead>
      </Container>
    </Wrapper>
  );
}

const Wrapper = styled.main`
  height: 75vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 320px;

  ${Container} {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`;
