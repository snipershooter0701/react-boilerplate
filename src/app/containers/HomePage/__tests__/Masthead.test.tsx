import React from 'react';
import renderer from 'react-test-renderer';
import { Masthead } from '../Masthead';

describe('<Masthead />', () => {
  it('should render and match the snapshot', () => {
    const renderedOutput = renderer.create(<Masthead />).toJSON();
    expect(renderedOutput).toMatchSnapshot();
  });
});
