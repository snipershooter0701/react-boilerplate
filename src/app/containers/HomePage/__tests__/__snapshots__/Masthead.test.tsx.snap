// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Masthead /> should render and match the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .logo {
  width: 4.5rem;
  height: 4.5rem;
}

.c1 .sign {
  width: 2rem;
  height: 2rem;
  margin: 0 2rem;
}

.c2 {
  font-size: 32px;
  font-weight: bold;
  margin: 1rem 0;
}

.c3 {
  font-size: 1.5rem;
  font-weight: 300;
  line-height: 1.5;
  margin: 0 0 1.5rem 0;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c4:active {
  opacity: 0.4;
}

.c0 {
  height: 60vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 320px;
}

<main
  className="c0"
>
  <div
    className="c1"
  >
    <IMAGE_MOCK
      className="logo"
    />
    <IMAGE_MOCK
      className="sign"
    />
    <IMAGE_MOCK
      className="logo"
    />
  </div>
  <h1
    className="c2"
  >
    React Boilerplate meets CRA
  </h1>
  <p
    className="c3"
  >
    Now you can use the
     
    <a
      className="c4"
      href="https://www.reactboilerplate.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      React Boilerplate
    </a>
     
    as a
     
    <a
      className="c4"
      href="https://github.com/facebook/create-react-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      Create React App
    </a>
     
    template.
  </p>
</main>
`;
